evalml/tests/data_checks_tests/test_invalid_targets_data_check.py

```diff
@@ -314,3 +320,67 @@ def test_invalid_target_data_check_initialize_with_none_objective():
     with pytest.raises(DataCheckInitError, match="Encountered the following error"):
         DataChecks([InvalidTargetDataCheck], {"InvalidTargetDataCheck": {"problem_type": "multiclass",
                                                                          "objective": None}})
+
+
+def test_invalid_target_data_check_regression_problem_nonnumeric_data():
```

Comment:
You could parametrize with a non-regression problem type to check the differences if you want to

Reply:
So I went ahead and parametrized it...I guess I'm not very familiar with what we ultimately want.  I know in some of my tests I use non-numerics for a categorical, but I don't think we're setup to do that, right?  ["Happy", "Birthday", "Birthday"] isn't considered a multiclass target, but [0, 1, 2, 2, 2, 1] is, right?